query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1030,9 +1184,20 @@ void ExecutionGenerator::convertCreateTable(
     catalog_relation->setDefaultStorageBlockLayout(layout.release());
   }
 
+  if (physical_plan->partition_scheme()) {
+    LOG(INFO) << "Relation created with partition scheme." << std::endl;
+    std::size_t num_partitions = physical_plan->partition_scheme()->getPartitionSchemeHeader().getNumPartitions();
+    attribute_id part_attr_id = physical_plan->partition_scheme()->getPartitionSchemeHeader().getPartitionAttributeId();
+    PartitionSchemeHeader *part_scheme_header = new HashPartitionSchemeHeader(num_partitions, part_attr_id);
```

Comment:
Yes, we support only Hash-based partitioning along with NUMA-support. We also don't have parser support for Range based partitioning. So, once we have that we can come back and add some conditions here to create range based partition schemes if needed.